src/lib.rs

```diff
@@ -813,50 +813,57 @@ use std::ffi::{CStr as StdCStr, CString as StdCString};
 
 #[cfg(feature = "std")]
 impl From<CString> for StdCString {
+    #[inline]
     fn from(s: CString) -> StdCString {
-        unsafe { StdCString::from_raw(s.into_raw()) }
+        unsafe { StdCString::from_vec_unchecked(s.into_bytes_with_nul()) }
```

Comment:
From the docs, `from_vec_unchecked` expects a `Vec` without a trailing nul byte. So actually we have to use `into_bytes` here instead.